src/modules/register/Register.jsx

```diff
@@ -233,17 +226,12 @@ const Form = styled.form`
 	}
 `;
 
-function Input(props) {
+function Input({type, name, autoComplete, onChange, value, className = '', children}) {
 	return (
-		<div className={'input-container' + props.class}>
-			<input
-				onChange={props.onChange}
-				type={props.type}
-				name={props.name}
-				id={props.name}
-				autoComplete={props.autoComplete}
-			/>
-			<label htmlFor={props.name}>{props.name}</label>
+		<div className={`input-container ${className}`}>
+			<input autoFocus onChange={onChange} type={type} name={name} id={name} autoComplete={autoComplete} value={value} />
```

Comment:
`autoFocus` is the fix to the bug you were running into (they should be on their own lines, I don't know why it appears inline here).

Here's the breakdown: You were calling `this.inputs.email.current.focus()` in the same tick that you were calling `this.setState({emailActive: true})`. This caused a situation where you hadn't actually called `render` yet, so `this.inputs.email` hadn't actually been attached as a `ref` yet, so `this.inputs.email.current` was `null`.

The solution is to take a wholly different approach: instead of having 3 different input widgets, where you carefully set `display: none` and apply `DOMElement::focus`, you should just only render a single input widget, and as the user advances through the flow, it updates the input widget, to render itself differently, i.e. as a name input, email or password.

The focus seems at first a little trickier, because we need to manage the `ref`, but we can actually just create a `ref` _within_ the input, and if we ensure that every time the screen changes, the input box is re-mounted, we can ensure that a new ref will be created. We can do that by adding a [`key`](https://reactjs.org/docs/lists-and-keys.html#keys) prop when we render `Input` that changes when the screen changes (I just used `this.state.active`).

Here's a class component that uses an internal ref to handle focus:
```jsx
class Input extends React.Component {
  constructor(props) {
    super(props)
    this.inputRef = React.createRef()
  }

  // this is guaranteed to only run after all refs have been mounted, 
  // and to only run once per component
  componentDidMount()  {
    this.inputRef.current.focus()
  }

  render() {
    ...
    return (
      <input
        ref={this.inputRef}
        ...
       />
    )
  }
}
```

But if you look at this, there's a lot of noise just to attach a `ref`. And we're only using the `ref` to focus the input box as soon as it gets mounted.

Well we're in luck because browsers support this use case out of the box. All we have to do is pass the `autoFocus` prop, and the widget will be focused the first time it mounts:
```jsx
function Input(props) {
  return (
    <input
      autoFocus
      ...
      />
    )
}
```